tests/integration/cloudformation/test_connection.py

```diff
@@ -114,9 +131,37 @@ def test_large_template_stack_size(self):
         self.assertEqual(None, policy)
 
         # Our new stack should show up in the stack list
-        stacks = self.connection.describe_stacks()
-        self.assertEqual(self.stack_name, stacks[0].stack_name)
-
-
+        stacks = self.connection.describe_stacks(self.stack_name)
+        stack = stacks[0]
+        self.assertEqual(self.stack_name, stack.stack_name)
+        
+        params = [(p.key, p.value) for p in stack.parameters]
+        self.assertEquals([('Parameter1', 'initial_value'),
+                           ('Parameter2', 'initial_value')], params)
+        
+        for _ in range(30):
+            stack.update()
+            if stack.stack_status.find("PROGRESS") == -1:
+                break
+            time.sleep(5)
+        
+        body = self.connection.update_stack(
+             self.stack_name,
+             template_body=json.dumps(BASIC_EC2_TEMPLATE),
+             parameters=[('Parameter1', '', True),
+                         ('Parameter2', 'updated_value')])
+        
+        stacks = self.connection.describe_stacks(self.stack_name)
+        stack = stacks[0]
+        params = [(p.key, p.value) for p in stacks[0].parameters]
+        self.assertEquals([('Parameter1', 'initial_value'),
+                           ('Parameter2', 'updated_value')], params)
+        
+        for _ in range(30):
```

Comment:
It's been quite a few months but I think it had to do with stack.update() responding immediately with a status of IN PROGRESS. So we need some kind of delay to keep the test going long enough for the update to actually take affect before attempting the update with the "previous value" in a second update. 

I'm open suggestions if there's a better way.